flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/source/assigner/SplitAssignerType.java

```diff
@@ -29,5 +29,10 @@ public SplitAssignerFactory factory() {
     }
   };
 
+  EVENTTIME {
```

Comment:
Something is missing here. I think this code is not compiling.
How do we plan to initialize the `EventTimeAwareSplitAssignerFactory`?

Reply:
I am mostly interested in the implementation of the `TimestampAssigner<IcebergSourceSplit>`. Especially for `CombinedScanTask`s

Reply:
> How do we plan to initialize the EventTimeAwareSplitAssignerFactory?

that is a good question. this may be a problem unless we use reflection and timestamp assigner class can be extracted from read config.

> TimestampAssigner<IcebergSourceSplit>. Especially for CombinedScanTasks

it will just look at the min value of the timestamp column using min-max column stats. for `CombinedScanTasks`, min should min of all files and max should be max of all files. But in practice, we probably want to avoid combining multiple files in a combined task. below is what we have in the internal unit test code.

```
        // essentially force one file per CombinedScanTask from split planning
        .splitOpenFileCost(256 * 1024 * 1024L);
```